### PR TITLE
Happy linting

### PR DESCRIPTION
**Pull request by ldemailly**
Fixing ineffectual assignment (slightly better).

Also exhibiting a build error

```release-note
NONE
```

### CODE REVIEW

**Comment by kyessenov** [comment]
fixed by https://github.com/istio/pilot/commit/a0a5dcdd66a0e62751a5e2eb393950b8487dbb47

**Comment by ldemailly** [comment]
this is a marginally better fix - interestingly it broke something

**Comment by rshriram** [comment]
closing temporarily. Sorry.

**Comment by ldemailly** [comment]
/retest
trying to debug build issues as a side effect of this, so please don't close
(it doesn't add significant load)

**Comment by ldemailly** [comment]
/test all

@yutongz seems to be stuck on that git error ( https://github.com/istio/test-infra/issues/428 ) any idea what may be the issue ?

**Comment by yutongz** [comment]
That's wired, I don't understand either. I did that merge on my machine and it works just fine. I am not a Prow expert but I guess you can try manually rebase from master so that prow doesn't need to.

**Comment by costinm** [comment]
Let me try again: can someone explain why the code fixed is wrong ? Is there any 
evidence that the linter is actually correct and there are tangible benefits of not
using perfectly valid go syntax ?

**Comment by gyliu513** [comment]
@ldemailly @yutongz I found the `make lint` has quite a few warning messages, do we need to fix all of them?

```
+ gometalinter --concurrency=2 --enable-gc --deadline=300s --disable-all --enable=aligncheck --enable=deadcode --enable=errcheck --enable=gas --enable=goconst --enable=gofmt --enable=goimports --enable=golint --exclude=.pb.go --exclude=gen_test.go --enable=gotype --enable=ineffassign --enable=interfacer --enable=lll --line-length=120 --enable=megacheck --enable=misspell --enable=structcheck --enable=unconvert --enable=varcheck --enable=vet --enable=vetshadow ./...
test/client/client.go:32:2:warning: error return value not checked (could not import github.com/gorilla/websocket (cannot find package "github.com/gorilla/websocket" in any of:) (errcheck)
test/client/client.go:111:35:warning: error return value not checked (undeclared name: websocket) (errcheck)
test/client/client.go:129:20:warning: error return value not checked (invalid operation: client (variable of type *invalid type) has no field or method Dial) (errcheck)
test/client/client.go:137:28:warning: error return value not checked (undeclared name: websocket) (errcheck)
test/client/client.go:237:14:warning: error return value not checked (undeclared name: websocket) (errcheck)
test/server/echo.go:40:2:warning: error return value not checked (could not import github.com/gorilla/websocket (cannot find package "github.com/gorilla/websocket" in any of:) (errcheck)
test/server/echo.go:58:16:warning: error return value not checked (undeclared name: websocket) (errcheck)
test/server/echo.go:40:2:warning: unused variable or constant could not import github.com/gorilla/websocket (cannot find package "github.com/gorilla/websocket" in any of: (varcheck)
test/server/echo.go:58:16:warning: unused variable or constant undeclared name: websocket (varcheck)
test/client/client.go:32:2:warning: unused variable or constant could not import github.com/gorilla/websocket (cannot find package "github.com/gorilla/websocket" in any of: (varcheck)
test/client/client.go:111:35:warning: unused variable or constant undeclared name: websocket (varcheck)
test/client/client.go:129:20:warning: unused variable or constant invalid operation: client (variable of type *invalid type) has no field or method Dial (varcheck)
test/client/client.go:137:28:warning: unused variable or constant undeclared name: websocket (varcheck)
test/client/client.go:237:14:warning: unused variable or constant undeclared name: websocket (varcheck)
test/client/client.go:32:2:warning: could not import github.com/gorilla/websocket (cannot find package "github.com/gorilla/websocket" in any of: (unconvert)
test/client/client.go:111:35:warning: undeclared name: websocket (unconvert)
test/client/client.go:129:20:warning: invalid operation: client (variable of type *invalid type) has no field or method Dial (unconvert)
test/client/client.go:137:28:warning: undeclared name: websocket (unconvert)
test/client/client.go:237:14:warning: undeclared name: websocket (unconvert)
test/server/echo.go:40:2:warning: could not import github.com/gorilla/websocket (cannot find package "github.com/gorilla/websocket" in any of: (unconvert)
test/server/echo.go:58:16:warning: undeclared name: websocket (unconvert)
test/server/echo.go:40:2:warning: could not import github.com/gorilla/websocket (cannot find package "github.com/gorilla/websocket" in any of: (interfacer)
test/server/echo.go:58:16:warning: undeclared name: websocket (interfacer)
test/client/client.go:32:2:warning: could not import github.com/gorilla/websocket (cannot find package "github.com/gorilla/websocket" in any of: (interfacer)
test/client/client.go:111:35:warning: undeclared name: websocket (interfacer)
test/client/client.go:129:20:warning: invalid operation: client (variable of type *invalid type) has no field or method Dial (interfacer)
test/client/client.go:137:28:warning: undeclared name: websocket (interfacer)
test/client/client.go:237:14:warning: undeclared name: websocket (interfacer)
test/client/client.go:32:2:warning: could not import github.com/gorilla/websocket (cannot find package "github.com/gorilla/websocket" in any of: (megacheck)
test/client/client.go:111:35:warning: undeclared name: websocket (megacheck)
test/client/client.go:129:20:warning: invalid operation: client (variable of type *invalid type) has no field or method Dial (megacheck)
test/client/client.go:137:28:warning: undeclared name: websocket (megacheck)
test/client/client.go:237:14:warning: undeclared name: websocket (megacheck)
test/server/echo.go:40:2:warning: could not import github.com/gorilla/websocket (cannot find package "github.com/gorilla/websocket" in any of: (megacheck)
test/server/echo.go:58:16:warning: undeclared name: websocket (megacheck)
test/server/echo.go:40:2:warning: could not import github.com/gorilla/websocket (cannot find package "github.com/gorilla/websocket" in any of: (aligncheck)
test/server/echo.go:58:16:warning: undeclared name: websocket (aligncheck)
test/client/client.go:32:2:warning: could not import github.com/gorilla/websocket (cannot find package "github.com/gorilla/websocket" in any of: (aligncheck)
test/client/client.go:111:35:warning: undeclared name: websocket (aligncheck)
test/client/client.go:129:20:warning: invalid operation: client (variable of type *invalid type) has no field or method Dial (aligncheck)
test/client/client.go:137:28:warning: undeclared name: websocket (aligncheck)
test/client/client.go:237:14:warning: undeclared name: websocket (aligncheck)
test/server/echo.go:40:2:warning: unused struct field could not import github.com/gorilla/websocket (cannot find package "github.com/gorilla/websocket" in any of: (structcheck)
test/server/echo.go:58:16:warning: unused struct field undeclared name: websocket (structcheck)
test/client/client.go:32:2:warning: unused struct field could not import github.com/gorilla/websocket (cannot find package "github.com/gorilla/websocket" in any of: (structcheck)
test/client/client.go:111:35:warning: unused struct field undeclared name: websocket (structcheck)
test/client/client.go:129:20:warning: unused struct field invalid operation: client (variable of type *invalid type) has no field or method Dial (structcheck)
test/client/client.go:137:28:warning: unused struct field undeclared name: websocket (structcheck)
test/client/client.go:237:14:warning: unused struct field undeclared name: websocket (structcheck)
Makefile:44: recipe for target 'lint' failed
make: *** [lint] Error 1
```

**Comment by costinm** [comment]
I think we should first agree on a style guide for Istio - including what
parts of Go language
are not allowed and why, and fix the linter checks instead.

**Comment by ldemailly** [comment]
@costinm (again) the linter's 'prefences' can be argued like any style, but the whole idea is to not argue. and in this specific case I think the code on the right is better than the earlier version, though not by much

I'm more interested in the build failure anyway for this

@yutongz afaik it's already rebased
